core/Segment.php

```diff
@@ -368,8 +370,23 @@ public function getSelectQuery($select, $from, $where = false, $bind = array(),
             $limitAndOffset = (int) $offset . ', ' . (int) $limit;
         }
 
-        return $this->segmentQueryBuilder->getSelectQueryString($segmentExpression, $select, $from, $where, $bind,
-            $groupBy, $orderBy, $limitAndOffset);
+        try {
+            if ($forceGroupBy && $groupBy) {
+                $this->segmentQueryBuilder->forceInnerGroupBySubselect(LogQueryBuilder::FORCE_INNER_GROUP_BY_NO_SUBSELECT);
```

Comment:
this was a feature I added recently... as we're using a `group by idvisit` we basically don't ever need to use a subselect...